docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -1,326 +1,752 @@
 
-* the Panache JPA extension
-* your JDBC driver extension (`quarkus-jdbc-postgresql`, `quarkus-jdbc-h2`, `quarkus-jdbc-mariadb`, ...)
+The Hibernate ORM with Panache extension contains the following classes:
 
-[source,xml]
+* `PanacheEntity`
+* `PanacheEntityBase`
+* `PanacheQuery`
+* `PanacheRepository`
+* `PanacheRepositoryBase`
+* `PanacheEntity`
+
+The {ProductName} Panache JPA extension eliminates the following Hiberante ORM issues:
+
+* *Duplicate ID logic*
++
+You use the `PanacheEntity` class to create entities. Because the ID is defined automatically in the `PanacheEntity` class, you do not need to define it in your `PanacheEntity` instance. However, you can specify your own ID in your instance if you prefer.  If you require a custom ID strategy, you can extend the `PanacheEntityBase` class instead.
+
+* *Useless `getter` and `setter` functions*
++
+Because Java does not support properties, when you write Java code you must create fields and then generate `getter` and `setter` functions for those fields, even if all you need to do is read or write to the field. With Panache, if you define a field as a public field and then read or write to that field, Panache generates the required `getter` and `setter` functions. In addition, you can write other accessors as required.
+* *Entity definitions split from operations*
++
+In the traditional Java EE model, you put the the entity definition in one class, the model, and the operations performed on the model in another class, the data access object (DAO) or repository. The Panache active record method enables you to create the entity definition and the repository in the same class.
+
+* *Verbose Hibernate queries*
++
+Panache provides Hibernate Query Language (HQL) shortcuts to simplify your queries. You do not write parts of the query that you do not need.
+
+
+
+
+=== Using the Panache active record pattern
+
+
+When you use Hiberate ORM, you must create two classes that separate the state and its operations even though you would never do that for regular Java objects where state and methods are in the same class. In addition to requiring two classes, Hibernate ORM requires you to  inject the repository where you need to perform entity operations, which breaks your edit flow and requires you to get out of the code that you are  writing to set up an injection point before coming back to use it.
+
+You can use the Panache active record pattern with the `PanacheEntity` class to keep the model and the repository together.
+
+.Procedure.
+
+. Add the Panache JPA extension dependency to your {ProductName} project `pom.xml` file:
++
+[source,java]
 ----
 <dependencies>
-    <!-- Hibernate ORM specific dependencies -->
     <dependency>
         <groupId>io.quarkus</groupId>
         <artifactId>quarkus-hibernate-orm-panache</artifactId>
     </dependency>
-
-    <!-- JDBC driver dependencies -->
```

Comment:
If you don't define a JDBC driver it won't work.

Reply:
The prerequisites for this guide, stated at the top:
 You have a Quarkus Maven project configured with a Java database connectivy (JDBC) datasource and Hibernate ORM. If they have a Quarkus Maven project that is configured with JDBC won't they have this code or similar in the pom.xml already?

    <!-- JDBC driver dependencies -->
    <dependency>
        <groupId>io.quarkus</groupId>
        <artifactId>quarkus-jdbc-postgresql</artifactId>
    </dependency>
</dependencies>

Reply:
Technically it's true, except I think that info is much more handy to duplicate here because users can cut and paste the whole thing.

Reply:
The problem with that is the example that is in the Panache guide is specific to Postgre. So to make it general takes a lot of explaining and anyway, the example is only useful is the user uses postgre. If I were to include it, I would have to provide a general template then the specific postgre example.

Reply:
Would it work if we just put in a warning that "if you don't define a jbdc driver things won't work" in addition to the prereq? I don't want to be repeating prereq information in the procedure.